File: mix/benchmark/fib.hs

import Control.Monad
import Text.Printf

fib :: Int -> Int
fib 0 = 0
fib 1 = 1
fib n = fib (n-1) + fib (n-2)

main = forM_ [0..34] $ \i ->
            printf "n=%d => %d\n" i (fib i)

File: mix/benchmark/fib.c

#include <stdio.h>

int fib(int n){
  if(n<2)
    return n;
  else
    return fib(n-1) + fib(n-2);
}

int main(){
  for(int i=0; i<35; ++i)
    printf("n=%d => %d\n", i, fib(i));
}
